### PR TITLE
Mount() is now in the Container object.

### LINKED ISSUE #9: Better detect overlay/overlay2 layouts
When processing an image that has both and "overlay2" and "overlay" folder it will take the "overlay2" folder by default even though there is no 'mounts' subfolder.

### PR DESCRIPTION
**Pull request by rgayon**
The Storage object which just handles mounting, disappears from de.py and is only contained in the Container object. Hence the 'storage detection' is now handled by the Container, as well as Mount().

This fixes #9 

the ShowRepositories() function is independent from Container objects, and is moved in de.py.

### CODE REVIEW

File: docker_explorer/lib/container.py [hunk 1]
```diff
@@ -204,3 +215,41 @@ def GetHistory(self, show_empty_layers=False):
...
+      storage_name (str): the name of the storage.
+    Returns:
+      BaseStorage: a storage object.
+    Raises:
+      BadContainerException: if no storage Driver is defined, or if it is not
+        implemented
+    """
+    storage_class = self.STORAGES_MAP.get(storage_name, None)
+
+    if storage_class is None:
```
Comment (tomchop) [inline docker_explorer/lib/container.py]: You can also do `if storage_name not in self.STORAGES_MAP`, and then assume that the key `storage_name` will always be valid (e.g. access it using `self.STORAGES_MAP[storage_name]`)

File: docker_explorer/lib/container.py [hunk 2]
```diff
@@ -204,3 +215,41 @@ def GetHistory(self, show_empty_layers=False):
...
+    """Mounts the specified container's filesystem.
+
+    Args:
+      mount_dir (str): the path to the destination mount point
+    """
+
+    commands = self.storage_object.MakeMountCommands(self, mount_dir)
+    for c in commands:
+      print(c)
+    print('Do you want to mount this container Id: {0:s} on {1:s} ?\n'
```
Comment (tomchop) [inline docker_explorer/lib/container.py]: typo: `container ID`

File: docker_explorer/lib/container.py [hunk 3]
```diff
@@ -204,3 +215,41 @@ def GetHistory(self, show_empty_layers=False):
...
+          '(ie: run these commands) [Y/n]'.format(self.container_id, mount_dir))
+    choice = raw_input().lower()
+    if not choice or choice == 'y' or choice == 'yes':
```
Comment (tomchop) [inline docker_explorer/lib/container.py]: `if not choice or or choice in ['y', 'yes']` or even `if choice in ['y', 'yes', '']`

File: docker_explorer/lib/container.py (hunk 1 above)
Comment (rgayon) [reply docker_explorer/lib/container.py]: I'll keep it as is, which looks better than:
`self.obj = self.STORAGES_MAP[storage_name](arg1, arg2)`

File: docker_explorer/lib/container.py (hunk 3 above)
Comment (rgayon) [reply docker_explorer/lib/container.py]: done

File: docker_explorer/lib/container.py (hunk 2 above)
Comment (rgayon) [reply docker_explorer/lib/container.py]: done

File: docker_explorer/lib/container.py (hunk 1 above)
Comment (tomchop) [reply docker_explorer/lib/container.py]: SG

**Pull request merged**